client/src/components/DataFiles/DataFilesModals/DataFilesUploadModalListing/DataFilesUploadModalListingTable.js

```diff
@@ -29,10 +32,15 @@ const DataFilesUploadStatus = ({ i, removeCallback }) => {
 };
 DataFilesUploadStatus.propTypes = {
   i: PropTypes.string.isRequired,
-  removeCallback: PropTypes.func.isRequired
+  removeCallback: PropTypes.func.isRequired,
+  rejectedFiles: PropTypes.arrayOf(PropTypes.object).isRequired
 };
 
-function DataFilesUploadModalListingTable({ uploadedFiles, setUploadedFiles }) {
+function DataFilesUploadModalListingTable({
+  uploadedFiles,
```

Comment:
if you refactor based on comment (link), then rejectedFiles and uploadedFiles have no overlap.  Then 
* DataFilesUploadStatus could be simplified 
* rejectedFiles could be displayed before the uploadedFiles in the list.

Which brings up a related question for you and design that is similar to a question I have in tickets:  If user alters the list of files (by selecting another file to upload etc), wouldn't we want to remove all the rejectedFiles as they just clutter the view?

Reply:
[Question asked in Slack.](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1620679749057800)

Reply:
Such a sorting is desirable. The sorting requirements can be ([source](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1620681932061400?thread_ts=1620679749.057800&cid=CQUA4D5KJ)):

1. rejected files
2. err’d during upload
3. successfully uploaded
4. not uploaded

I suggest this __not__ be a Design requirement for this PR / FP-250.

Reply:
_But_, any existing sorting __should not__ be lost because of changes in this PR.

Reply:
I am still unsure about whether files should be removed from rejected files if a file of the same name is tried again. [See question to Design.](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1620683525062300?thread_ts=1620679749.057800&cid=CQUA4D5KJ)

Reply:
**Yes**—confirmed—files should be removed from rejected files if a file of the same name is tried again.

Reply:
Instead of **same name** we'll do this for **same path**.